File: backend-nestjs/patient-service/src/management-suggestion/management-suggestion.entity.ts
import { Entity, Column, PrimaryColumn } from 'typeorm';

@Entity('management_suggestions')
export class ManagementSuggestionEntity {
  @PrimaryColumn({ type: 'bigint' })
  id: number;

  @Column({ name: 'tenant_id', type: 'bigint' })
  tenantId: number;

  @Column({ name: 'org_id', type: 'bigint' })
  orgId: number;

  @Column({ name: 'patient_id', type: 'bigint' })
  patientId: number;

  @Column({ name: 'created_by_user_id', type: 'bigint' })
  createdByUserId: number;

  @Column({ name: 'suggestion_type' })
  suggestionType: string;

  @Column({ name: 'content' })
  content: string;

  @Column({ name: 'status' })
  status: string;

  @Column({ name: 'created_at', type: 'timestamp' })
  createdAt: Date;
}

File: backend-java/cdm-patient/src/main/java/com/cdm/patient/entity/ManagementSuggestionEntity.java
package com.cdm.patient.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.cdm.common.domain.BaseEntity;
import com.cdm.patient.vo.ManagementSuggestionVo;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@TableName("management_suggestions")
@Getter
@Setter
@NoArgsConstructor
public class ManagementSuggestionEntity extends BaseEntity {

    private Long tenantId;
    private Long orgId;
    private Long patientId;
    private Long createdByUserId;
    private String suggestionType;
    private String content;
    private String status;
    private LocalDateTime createdAt;

    public static ManagementSuggestionVo toVo(ManagementSuggestionEntity entity) {
        if (entity == null) return null;
        return ManagementSuggestionVo.builder()
                .id(entity.getId())
                .tenantId(entity.getTenantId())
                .orgId(entity.getOrgId())
                .patientId(entity.getPatientId())
                .createdByUserId(entity.getCreatedByUserId())
                .suggestionType(entity.getSuggestionType())
                .content(entity.getContent())
                .status(entity.getStatus())
                .createdAt(entity.getCreatedAt())
                .build();
    }
}
